packages/core/src/PartialExecutionUtils/findTriggerForPartialExecution.ts

```diff
@@ -0,0 +1,56 @@
+import type { INode, Workflow } from 'n8n-workflow';
+
+function findAllParentTriggers(workflow: Workflow, destinationNode: string) {
+	const parentNodes = workflow
+		.getParentNodes(destinationNode)
+		.map((name) => {
+			const node = workflow.getNode(name);
+
+			if (!node) {
+				return null;
+			}
+
+			return {
+				node,
+				nodeType: workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion),
+			};
+		})
+		.filter((value) => value !== null)
+		.filter(({ nodeType }) => nodeType.description.group.includes('trigger'))
+		.map(({ node }) => node);
+
+	return parentNodes;
+}
+
+// TODO: write unit tests for this
+export function findTriggerForPartialExecution(
+	workflow: Workflow,
+	destinationNode: string,
+): INode | undefined {
+	const parentTriggers = findAllParentTriggers(workflow, destinationNode).filter(
+		(trigger) => !trigger.disabled,
+	);
+	const pinnedTriggers = parentTriggers
+		// TODO: add the other filters here from `findAllPinnedActivators`, see
+		// copy below.
+		.filter((trigger) => workflow.pinData?.[trigger.name])
+		.sort((n) => (n.type.endsWith('webhook') ? -1 : 1));
```

Comment:
Well, I copied that code from the old flow. And I'm going to document, but I'm also going to revise it later, because this puts all nodes that end with `webhook` first, but reverses their previous position 😄 

![image](https://github.com/user-attachments/assets/c3883704-95e7-4f0a-a76a-bc6f4410ff09)